### PR TITLE
[quant][core][gpu][improvement] Removed conv_output and set output tensors as virtual in quantized cudnn conv2d op

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76788
* __->__ #76787
* #76518

Summary:
With support for virtual tensors in cudnn, we no longer have to allocate
conv_output.

Test plan:
```
python test/test_quantization.py -k test_qconv2d_cudnn
```

Differential Revision: [D36121583](https://our.internmc.facebook.com/intern/diff/D36121583)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36121583).

**Review by jerryzh168 — APPROVED** [review]

File: aten/src/ATen/native/quantized/cudnn/Conv.cpp [hunk 1]
```diff
@@ -189,7 +177,8 @@ void PackedConvWeightCudnn<kSpatialDim>::apply_impl_helper(const at::Tensor& qua
   // output is a fp32 tensor
   auto conv_op = cudnn_frontend::OperationBuilder(CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR)
       .setxDesc(cudnn_utils::getTensorDescriptor(input.sizes(), input.strides(), CUDNN_DATA_INT8, 'x', key.input_alignment))
-      .setyDesc(cudnn_utils::getTensorDescriptor(conv_output, 'y', key.output_alignment))
+      // for virtual tensors, the alignment is not used, so we can just put an arbitrary value here, e.g., key.output_alignment
+      .setyDesc(cudnn_utils::getTensorDescriptor(quantized_output.sizes(), quantized_output.strides(), CUDNN_DATA_FLOAT, 'y', key.output_alignment, true))
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/Conv.cpp:181]: should we use a constant here?

File: aten/src/ATen/native/quantized/cudnn/Conv.cpp [hunk 2]
```diff
@@ -136,7 +135,7 @@ void PackedConvWeightCudnn<kSpatialDim>::apply_impl_helper(const at::Tensor& qua
   // set the datatype for output tensor to int32 or fp32
   key.params.dataType = CUDNN_DATA_INT32;
   key.input_alignment = cudnn_utils::getAlignment(input);
-  key.output_alignment = cudnn_utils::getAlignment(conv_output);
+  key.output_alignment = cudnn_utils::getAlignment(quantized_output);
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/Conv.cpp:138]: looks like this is not useful, maybe we can use a constant

File: aten/src/ATen/native/quantized/cudnn/Conv.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/quantized/cudnn/Conv.cpp:181]: this `setyDesc` is not really used since the output is a virtual tensor, is that corrrect?

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36121583).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36121583).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36121583).

**Pull request closed without merging**